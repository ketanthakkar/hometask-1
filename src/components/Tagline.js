import React from 'react';
import '../styleModules/Tagline.css';

const Tagline = ({tagline}) => <p className = 'tagline'> {tagline} </p>;

export default Tagline;
