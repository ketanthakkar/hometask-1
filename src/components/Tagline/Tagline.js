import React from 'react';
import './Tagline.css';

const Tagline = ({tagline}) => <p className = 'tagline'> {tagline} </p>;

export default Tagline;
