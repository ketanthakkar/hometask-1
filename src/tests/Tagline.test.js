import React from 'react';
import { shallow } from 'enzyme';
import Tagline from '../components/Tagline';

describe('Tagline', () => {
  it('shoud render Tagline component', () => {
    const render = shallow(
      <Tagline tagline = 'test' />
    );
    expect(render.find('p').hasClass('tagline')).toBe(true);
    expect(render).toMatchSnapshot();
  })

  it('should not modify props', () => {
    const tagline = 'test';
    const render = shallow(
      <Tagline tagline = {tagline} />
    );
    expect(tagline).toEqual('test');
    expect(render).toMatchSnapshot();
  })
})