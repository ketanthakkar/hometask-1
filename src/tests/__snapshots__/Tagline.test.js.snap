// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tagline shoud render Tagline component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tagline
    tagline="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        "test",
        " ",
      ],
      "className": "tagline",
    },
    "ref": null,
    "rendered": Array [
      " ",
      "test",
      " ",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "test",
          " ",
        ],
        "className": "tagline",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "test",
        " ",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Tagline should not modify props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tagline
    tagline="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        "test",
        " ",
      ],
      "className": "tagline",
    },
    "ref": null,
    "rendered": Array [
      " ",
      "test",
      " ",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "test",
          " ",
        ],
        "className": "tagline",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "test",
        " ",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
